k2/csrc/fsa_algo.cu

```diff
@@ -141,6 +141,16 @@ bool Intersect(FsaOrVec &a_fsas, FsaOrVec &b_fsas,
     }
     // the check on the previous line will fail.
   }
+  if (properties_a < 0) {
+    properties_a = GetFsaBasicProperties(a_fsas);
+  }
+  if (properties_b < 0) {
+    properties_b = GetFsaBasicProperties(b_fsas);
+  }
+  bool check_properties = true;
+  if ((properties_a & kFsaPropertiesArcSorted) && (properties_b & kFsaPropertiesArcSorted)) {
+    check_properties = false;
+  }
```

Comment:
@danpovey 
should we return early here if one of the input FSAs is not arc sorted?
If an FSA is not arc sorted, there is no need to call k2host::Intersect
as we know it will return an empty FSA.

Reply:
I was thinking we would always call it with check_properties = false (from here).
Since we have called GetFsaBasicProperties() we know the properties.
It will be an error if they are not arc-sorted, i.e. we will die with error message.

Reply:
you mean, remove this if block and always pass check_properties = false?

Reply:
or use ASSERT_EQ for the properties?

Reply:
I guess this is what Dan meant:
```cpp
K2_CHECK_EQ(check_properties, false) << "some log message";
```

Reply:
OK, done

Reply:
@csukuangfj this check failed in two tests. Not familiar with the error message. Not sure if it is because of some un-arc-sorted input fsa...